CONTRIBUTING.md

```diff
@@ -0,0 +1,139 @@
+All processes in this document are designed to streamline the development effort, to avoid bottlenecks, and to ultimately give a pleasant experience to all involved.
+
+Architecture and Scope of Cocotb
+--------------------------------
+
+Cocotb has seen adoption in a wide variety of scenarios with sometimes conflicting requirements.
+To foster experimentation and to decentralize the development process the architecture of cocotb is highly modular.
+A solid core forms the foundation upon which extensions can provide higher-level functionality.
+
+The core of cocotb are
+- the infrastructure to write testbenches with coroutines, threads, etc.,
+- the abstraction and interaction with simulators through interfaces like VPI, GPI, etc.,
+- tooling to run tests, and
+- core primitives to interact with the simulation: triggers, data access classes, etc.
+
+As a general rule, functionality beyond this core set should go into extensions.
+However, none of these rules are set in stone.
+They can and should be challenged at times to ensure the project stays relevant to the majority of its users.
+
+
+How to Get Changes Merged
+-------------------------
+
+Have you fixed a bug in cocotb, or want to add new functionality to it?
+Cocotb follows the typical [GitHub flow](https://guides.github.com/introduction/flow/) and makes use of pull requests and reviews.
+More specificially, follow the steps below to get your changed merged, i.e. integrated into the main cocotb codebase.
+
+1. Create an issue ticket on [cocotb's GitHub issue tracker](https://github.com/potentialventures/cocotb/issues) describing your problem.
+   This step is optional for smaller changes, but highly recommended for larger ones.
+   Issues are a good place to discuss design options with others before writing code.
+2. [Fork](https://help.github.com/articles/fork-a-repo/) the [cocotb GitHub repository](https://github.com/potentialventures/cocotb) into your personal namespace.
+3. Create a new branch off the `master` branch for your set of changes.
+   Use one branch per "topic," i.e. per set of changes which belong together.
+4. Create one or multiple commits to address the issue.
+   Make sure to read and follow the [Patch Requirements](#patch-requirements) when preparing your commits.
+5. Create new [pull request (PR)](https://github.com/potentialventures/cocotb/pulls).
+6. When you submit (or update) the pull request, a suite of regression tests will run.
+   If any of them turns "red," i.e. reports a failure, you most likely need to fix your code before it can be merged.
+7. The pull request needs to be reviewed by at least one maintainer.
+   We aim to give feedback to all pull requests within a week, but as so often, life can get in the way.
+   If you receive no feedback from a maintainer within that time, please contact him/her directly (e.g. on IRC or email). You can find a [list of all maintainers](#maintainers) and their main area of expertise [below](#maintainers).
```

Comment:
We don't have IRC (right?), but we could use Gitter.

Reply:
good catch, I've added the Gitter link.